torch/distributed/_sharded_tensor/ops/linear.py

```diff
@@ -0,0 +1,129 @@
+import torch
+import torch.distributed as dist
+from torch.distributed._sharding_spec import ChunkShardingSpec
+from torch.distributed._sharding_spec._internals import (
+    get_split_size,
+    get_chunked_dim_size,
+)
+from typing import List
+
+
+def sharded_linear(types, args, kwargs, pg):
+    from torch.distributed._sharded_tensor import ShardedTensor
+
+    input = args[0]
+    weight = args[1]
+    bias = kwargs["bias"]
+
+    # Validate types
+    if not isinstance(input, torch.Tensor) or not isinstance(bias, torch.Tensor):
+        raise TypeError("input and bias need to be torch.Tensor")
+    if not isinstance(weight, ShardedTensor):
+        raise TypeError("weight needs to be ShardedTensor")
+
+    if input.size()[1] != weight.size()[1]:
+        raise ValueError(
+            f'Input dim: {input.size()[1]} does not match '
+            f'appropriate weight dim: {weight.size()[1]}')
+    if not isinstance(weight._sharding_spec, ChunkShardingSpec):
+        raise ValueError("Only ChunkShardingSpec supported for ShardedTensor ops!")
+    if len(weight.local_shards()) != 1:
```

Comment:
nit: one of two test cases for validation error? 

It seems you are testing ops/linear.py in test_sharded_tensor.py 

Probably it can be useful to add test_linear as well since we might need a few unit tests for it since it includes a lot of new logics.

Reply:
I meant to have some local test like test_linear.py 

basically each py file like {foo}.py has a test_{foo}.py